pkg/bgpv1/gobgp/state.go

```diff
@@ -281,3 +281,49 @@ func (g *GoBGPServer) GetRoutes(ctx context.Context, r *types.GetRoutesRequest)
 		Routes: routes,
 	}, nil
 }
+
+// GetRoutePolicies retrieves route policies from the underlying router
+func (g *GoBGPServer) GetRoutePolicies(ctx context.Context) (*types.GetRoutePoliciesResponse, error) {
+	// list defined sets into a map for later use
+	definedSets := make(map[string]*gobgp.DefinedSet)
+	err := g.server.ListDefinedSet(ctx, &gobgp.ListDefinedSetRequest{DefinedType: gobgp.DefinedType_NEIGHBOR}, func(ds *gobgp.DefinedSet) {
+		definedSets[ds.Name] = ds
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed listing neighbor defined sets: %w", err)
+	}
+	err = g.server.ListDefinedSet(ctx, &gobgp.ListDefinedSetRequest{DefinedType: gobgp.DefinedType_PREFIX}, func(ds *gobgp.DefinedSet) {
+		definedSets[ds.Name] = ds
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed listing prefix defined sets: %w", err)
+	}
+
+	// list policy assignments into a map for later use
+	assignments := make(map[string]*gobgp.PolicyAssignment)
+	err = g.server.ListPolicyAssignment(context.Background(), &gobgp.ListPolicyAssignmentRequest{}, func(a *gobgp.PolicyAssignment) {
+		for _, p := range a.Policies {
+			assignments[p.Name] = a
+		}
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed listing policy assignments: %w", err)
+	}
+
+	// list & convert policies
+	var policies []*types.RoutePolicy
+	err = g.server.ListPolicy(ctx, &gobgp.ListPolicyRequest{}, func(p *gobgp.Policy) {
+		assignment := assignments[p.Name]
```

Comment:
nit : use map lookup boolean field for existence check.
```
assignment, exists := assignments[p.Name]
if exists {
  ...
}
```

assignment is point type, but if someday we change its type to non-pointer, assignment != nil will always be true.

Reply:
fixed in https://github.com/cilium/cilium/pull/28973/commits/b88fa8491d5e4335417dac7a3d216039b7ca8c43